docs/cloud-edge/modules/oauth.mdx

```diff
@@ -0,0 +1,700 @@
+
+### Special Paths
+
+Upstream applications behind endpoints with this module enabled do not receive
+any requests to paths beginning with `/auth/`. Your application may redirect
+clients to the following paths to invoke different behaviors.
+
+| Path           | Description                                                                                                                                                                                                                                                                                                                    |
+| -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `/auth/authn`  | Redirect users to this path to explicitly begin an authentication flow. After authentication, users will be redirected to `/`. If the IdP supports it, ngrok will attempt to instruct the IdP to force reauthentication which will force users to re-enter their credentials with the IdP even if they were already logged in. |
+| `/auth/logout` | Logs the user out by clearing their session cookie. Redirect users to this path to log them out.                                                                                                                                                                                                                               |
+
+### Events
+
+When this module is enabled, it populates the following fields in the
+[http_request_complete.v0](/events/reference/#http-request-complete) event:
+
+| Fields                |
+| --------------------- |
+| `oauth.app_client_id` |
+| `oauth.decision`      |
+| `oauth.user.id`       |
+| `oauth.user.name`     |
+
+### Errors
+
+:::note
+
+This documentation is incomplete at this time. We're working to improve it with
+a full list of errors.
+
+:::
+
+### Limits
+
+#### Managed Applications {#managed-application-limits}
+
+[Managed OAuth applications](#managed-applications) are owned by ngrok and have
+limitations to prevent abuse.
+
+- With a managed oauth application you may not pass custom scopes.
+- The [upstream headers](#upstream-headers) `ngrok-auth-oauth-access-token` and
+  `ngrok-auth-oauth-refresh-token` are not sent to your application.
+
+#### MAUs {#mau-limits}
+
+MAU stands for "Monthly Active User". Monthly Active Users are the number of
+uniquely authenticated OAuth users who have accessed your endpoints within a
+month.
+
+OAuth MAU limits are enforced account-wide, they are _not_ enforced on a
+per-endpoint basis.
+
+An authenticated user with the same ID from the identity provider is counted as
+a single MAU even if they connect to multiple endpoints on your account.
+
+| Plan       | MAUs                            |
```

Comment:
should we document these limits here, or redirect users to the pricing page. i'm just thinking about future maintenence.

Reply:
probably. but right now there's no way to link to a specific line in the pricing table so our only option is to dump you on the pricing page at the top which feels really bad. i figure we can revisit this with the next rev of the pricing page and take this into consideration as part of the design. cc @samcrichard